### PR TITLE
test(web): raise coverage for admin, auth, billing, and shared utilities

### PR DESCRIPTION
**Pull request by ZappoMan**
## Summary
- Raises web test coverage for admin UI, auth flows, billing pages, hooks, and shared lib utilities to meet ≥99% statements, ≥95% branches, ≥98% functions, and ≥98% lines on the targeted files.
- Adds targeted tests for error paths, storage/SSR edge cases, and conditional UI branches; includes small refactors to remove unreachable defensive code.
- Makes billing plan CTAs omit no-op `onClick` handlers when disabled (current plan / loading states).

## Test plan
- [x] `cd apps/web && npm exec -- vitest run`
- [x] `cd apps/web && npm exec -- vitest run --coverage`
- [ ] CI checks pass on this PR

### CODE REVIEW

File: apps/web/src/__tests__/useMarketingAuthHint.test.tsx [hunk 1]
```diff
@@ -10,6 +10,10 @@ describe('useMarketingAuthHint', () => {
     vi.clearAllMocks();
   });
 
+  afterEach(() => {
+    vi.unstubAllGlobals();
+  });
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/web/src/__tests__/useMarketingAuthHint.test.tsx]: Fixed.

File: packages/admin/src/components/AdminDetailDrawer.web.tsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ export function AdminDetailDrawer({
       if (focusables.length === 0) return;
 
       const first = focusables[0];
-      const last = focusables[focusables.length - 1];
+      const last = focusables.at(-1);
       if (first === undefined || last === undefined) return;
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/admin/src/components/AdminDetailDrawer.web.tsx:59]: Addressed in 371de26 — reverted to ES2020-safe indexing:

```ts
const last = focusables[focusables.length - 1];
```

This matches the package TS target/lib (ES2020) and avoids relying on `Array.prototype.at`. The drawer focus-trap tests still pass.
